### ISSUE TITLE
Feature Request: Send notification email that subscription delay is ending

### ISSUE THREAD

**Issue by andrewlimaza** [strangerstudios/pmpro-subscription-delays#9]
A user would like the ability to send an email before the subscription delay ends, similar to trial end email template.

**Comment by ideadude** [comment]
Yes. Let's do this. We can use this email template for reference:
https://github.com/strangerstudios/paid-memberships-pro/blob/dev/email/trial_ending.html

Note that we don't actually send this email in the core PMPro plugin anymore (because we cheat and setup trials to separate out the first payment). Here it's okay to send this email. We should be able to figure out the date to send this... maybe we need to set a user meta at checkout and keep track of that in case the level/sub changes.

**Comment by ideadude** [comment]
So we can use this user meta:
https://github.com/strangerstudios/pmpro-subscription-delays/blob/master/pmpro-subscription-delays.php#L144

Probably need to make sure that that gets cleared out if the level changes still.
